ActionPlans/Start-DlToO365GroupUpgradeChecks.ps1

```diff
@@ -303,45 +303,61 @@ else {
 }
 #endregion Check if Distribution Group can't be upgraded because Distribution list which has more than 100 owners or it has no owner
 
-#new condition to check if owner has mailbox or not
-#region Check if Distribution Group can't be upgraded because anyone of the owners doesn't have a mailbox
+#region Check if Distribution Group can't be upgraded because the distribution list owner(s) is non-supported with RecipientTypeDetails other than UserMailbox, MailUser
 if ($checkifownerhasmailbox -match "Continuechecking")
 {
-    [string]$SectionTitle = "Validating Distribution Group Owners Mailbox Status"
-    [string]$Description = "Checking if Distribution Group can't be upgraded because one or more DL owners doesn't have a mailbox in Exchange Online"
-    $ConditionDGownerswithoutMBX=@()
+    [string]$SectionTitle = "Validating Distribution Group Owners RecipientTypeDetails"
+    [string]$Description = "Checking if Distribution Group can't be upgraded because DL owner(s) is non-supported with RecipientTypeDetails other than UserMailbox, MailUser"
+    $ConditionDGownernonsupported=@()
+    $ConditionDGownernonsupportedforusers=@()
     foreach($owner in $owners)
     {
         try {
-            $owner=Get-User $owner -ErrorAction stop
-            if ($owner.RecipientType -eq "User") 
+            $owner=Get-Recipient $owner -ErrorAction stop
+            if ($owner.RecipientTypeDetails -ne "UserMailbox" -and $owner.RecipientTypeDetails -ne "MailUser") 
```

Comment:
this condition will always be false
even for MailUser as is not UserMailbox and vice-versa
(as it cannot be in the same time UserMailbox  and MailUser )
Maybe you need to change the AND with OR

Reply:
I've changed that part as you recommended over Teams to be 

 if ($owner.RecipientTypeDetails -eq "UserMailbox" -or $owner.RecipientTypeDetails -eq "MailUser") 
                { 
                   #Do Nothing
                }
                else{
                    $ConditionDGownernonsupported=$ConditionDGownernonsupported+$owner
                }

Reply:
we cleared the condition by doing negation